### DISCUSSION TITLE
Anyway to use this while also filming on the iphone or ipad?

### DISCUSSION THREAD

**Discussion by RikardSwed** [lumetrium/obsidian-teleprompter#20] [Q&A]
Hi

Is there a way to use this telepromper while filming? perhaps like some telepromper apps can float while you use the camera?
Is is possible already or might there be a work around? Is it perhaps possible to add this feature?

What kind of setup do you use to get the most out of this app and create high quality videos?

thanks for the help!

**Comment by lumetrium** [comment]
Hi! I don't have access to an iPhone or iPad, so I can't say if it's possible at the OS level. However, I do have an Android tablet, specifically a Samsung Galaxy Tab S series with OneUI, and I can easily place the Obsidian app with the teleprompter over the camera using the "Open in pop-up view" feature in OneUI:

https://github.com/lumetrium/obsidian-teleprompter/assets/125469344/460fb59f-7a79-485a-9938-63e39ce5133d


Regarding the setup, I personally use the teleprompter on my PC. The two main use cases for which I use it are:

1. **Recording myself for a video**
    - I place the Obsidian window on a separate monitor and position the teleprompter tab directly underneath the camera. Then, I start the recording and the teleprompter using the Stream Deck.

2. **Talking on a live call**
    - I detach the teleprompter so it's in a separate window from the Obsidian app. Then, I place it over the window with the call app and reduce the opacity to around 60-70%. This way, I can see my audience, myself, and the running text.

In all of this, I rely heavily on the Stream Deck. I use the "Website" action under the "System" category to execute commands via the [Advanced URI](https://github.com/Vinzent03/obsidian-advanced-uri) plugin for Obsidian. For instance, the URL to toggle the play/stop function of the teleprompter looks like this:
```
obsidian://advanced-uri?vault=Master&commandid=teleprompter:control:play:toggle
```

I have many buttons for various teleprompter commands. I might share the Stream Deck profile with everyone after I polish it a bit.

I should mention that my use of the teleprompter is fairly casual. I mainly use it for making simple videos or avoiding embarrassment on live calls where I need to talk a lot. I'm not a professional video creator, so my setup is geared more towards personal ease and efficiency.

**Comment by RikardSwed** [comment]
Sounds cool with the stream deck possibilites. 

I wish I could the camera on my iphone as it is much better than the one on my macbook air. 
Is OneUI an app? I dont know how to get obsidian to be minimized like that.
I can do it with apps like youtube by just pressing the home button while youtube is playing but other than that I don't think its possible.

**Reply by lumetrium** [reply]
One UI is a version of Android customized by Samsung for its devices.

I looked it up and found that a similar functionality exists on iPads called [Slide Over](https://www.youtube.com/watch?v=HyD_y28qkAg). It seems pretty limited though, with no way to resize the "slide over" window. On iPhones, I'm pretty sure this is impossible.

What you're describing with YouTube is called "Picture in Picture" or PiP for short. As I understand it, on iOS, PiP is specific to media playback and doesn't support broader interactivity or custom user interfaces. 

Even then, app developers must implement support for it themselves. This teleprompter is not a standalone app but a plugin for the Obsidian app, and it doesn't render video but shows an HTML page with text, scrolling it with JavaScript. In other words, PiP won't work in this case.

As an alternative, I suggest you try using your iPhone as a video input device for your MacBook. I found this official guide that explains how to do it: https://support.apple.com/guide/mac-help/use-iphone-as-a-webcam-mchl77879b8a/mac

This way, you can both record videos and use the teleprompter on your Mac while shooting on your iPhone.
